spyder/app/mainwindow.py

```diff
@@ -1072,13 +1073,27 @@ def create_edit_action(text, tr_text, icon):
             section=ToolsMenuSections.Tools,
             before=winenv_action
         )
-        if get_debug_level() >= 3:
-            self.menu_lsp_logs = QMenu(_("LSP logs"))
-            self.menu_lsp_logs.aboutToShow.connect(self.update_lsp_logs)
+
+        # Debug logs
+        if get_debug_level() >= 2:
+            self.menu_debug_logs = QMenu(_("Debug logs"))
+            self.menu_debug_logs.aboutToShow.connect(self.update_debug_logs)
             mainmenu.add_item_to_application_menu(
-                self.menu_lsp_logs,
+                self.menu_debug_logs,
                 menu_id=ApplicationMenus.Tools)
 
+        # Restart in debug mode
+        restart_in_debug_mode_action = create_action(
+            self,
+            _("Restart in DEBUG mode"),
+            None, icon=ima.icon('restart'),
+            triggered=self.restart_in_debug_mode,
+            tip=_("Restart in DEBUG mode"))
+        mainmenu.add_item_to_application_menu(
+            restart_in_debug_mode_action,
+            menu_id=ApplicationMenus.Tools
+        )

```

Comment:
Please move this action and the code to add it to the Tools menu to the Application plugin (i.e. to `plugins/application/container.py`).

Reply:
Perhaps this should be in the File menu, beneath "Restart", instead of the Tools menu?

Reply:
I was thinking that too and it makes a lot of sense.

@dalthviz, @isabela-pf, what do you think?

Reply:
So the idea is to add a "Restart in debug mode" entry either to the File or Tools menu to allow Spyder to print a lot of debugging diagnostics.

Reply:
That's right. In particular, for non-expert users of the macOS application, this will make it much easier to produce debug output to share with developers.

Reply:
I may need some tutelage on this. I'm confused about how this is supposed to work. I'm looking at the Restart stuff as an example but can't quite make sense of it.
So, the menu item for Restart is created in application/plugin.py and has a callable action assigned, `Application.restart_action`. `Application.restart_action` is just a wrapper to `ApplicationContainer.restart_action` in application/container.py. `ApplicationContainer.restart_action` has a callable, `PluginMainContainer.sig_restart_requested`, which is just a `Signal`. But I can't find that signal emitted anywhere. 
To further confuse things, the `Application` plugin in application/plugin.py also has a signal inherited from `SpyderPluginV2.sig_restart_requested`.

It seems to me, in my naive understanding, that if we want the menubar items created in the `Application` plugin, then the menubar item should just be attached to the `Applicationi.restart` method directly, or it should emit the `sig_restart_requested` (`PluginMainContainer.sig_restart_requested` or `SpyderPluginV2.sig_restart_requested`?) signal and the signal should be connected to `Application.restart`.  Am I misunderstanding something?

Reply:
@mrclary yep can be quite confusing but basically all the plugins have the signal to requested a restart (event the Application plugin) but the actual plugin that implements the functionality is the Application plugin. Then what we are doing is using the mainwindow as the central receptor of this signals (for plugins and plugin containers I think) but at the end what the mainwindow does to handle the request is to ask the Application plugin to do the restart.

Also, inside the plugin (the plugin vs container relation), I think we ended up doing it like that to keep all the actions definitions in the same place (in this case the container). If we want to still keep all the actions in the  container then the new action should also trigger a `sig_restart_requested` but passing the debug flag (then we need also to add a new signature for the `sig_restart_requested` signal for SpyderV2 and PluginMainContainer ) or probably create a new signal that triggers a restart on debug mode (and add the logic to handle the new signal in a similar way as what is done for the `sig_restart_requested`).

The other option is to move the creation of the action to the plugin and directly use as trigger the restart method (one action will use the flag to call the restart with the debug flag an the other just the default restart).

Reply:
> @mrclary yep can be quite confusing but basically all the plugins have the signal to requested a restart (event the Application plugin) but the actual plugin that implements the functionality is the Application plugin. Then what we are doing is using the mainwindow as the central receptor of this signals (for plugins and plugin containers I think) but at the end what the mainwindow does to handle the request is to ask the Application plugin to do the restart.

Yes, I see that. It was confusing since the mainwindow actually handled the menu item and the menu item did not reference the signal, just called `self.restart`->`Application.restart`. While it connects the signal to `self.restart`, the signal is never emitted anywhere, so is never used. By moving the menu handling to the `Application` plugin, we would would also want to remove any reference to `sig_restart_requested` from mainwindow, since the signal would be handled in the plugin.

> Also, inside the plugin (the plugin vs container relation), I think we ended up doing it like that to keep all the actions definitions in the same place (in this case the container). If we want to still keep all the actions in the container then the new action should also trigger a `sig_restart_requested` but passing the debug flag (then we need also to add a new signature for the `sig_restart_requested` signal for SpyderV2 and PluginMainContainer ) or probably create a new signal that triggers a restart on debug mode (and add the logic to handle the new signal in a similar way as what is done for the `sig_restart_requested`).
> 
> The other option is to move the creation of the action to the plugin and directly use as trigger the restart method (one action will use the flag to call the restart with the debug flag an the other just the default restart).

We are already using the container paradigm, so I'll stick with that and add a signature for `sig_restart_requested`; but I'm still confused as to why we need `sig_restart_requested` defined in two places. Shouldn't we be trying to have a single source for the definition? Also, doesn't this make two separate `Signal` instances, so that it matters which one the slot is connected to? In other words, slots connected to `SpyderPluginV2.sig_restart_requested` don't get executed if `PluginMainContainer.sig_restart_requested` is emitted?